src/layers/domain/core/device.py

```diff
@@ -0,0 +1,351 @@
+import re
+from enum import StrEnum, auto
+from uuid import UUID
+
+from domain.core.accredited_system_id import AccreditedSystemId
+from domain.core.aggregate_root import AggregateRoot
+from domain.core.error import (
+    DuplicateError,
+    InvalidDeviceKeyError,
+    InvalidKeyError,
+    NotFoundError,
+)
+from domain.core.event import Event
+from domain.core.product_id import ProductId
+from pydantic import BaseModel, Field
+
+
+class DeviceCreatedEvent(Event):
```

Comment:
I've rewritten these as native python dataclasses to avoid hand rolled data missing fields on serialising to dynamodb